client/pluginManger.go

```diff
@@ -133,6 +139,48 @@ func (p *PluginManager) InstallPlugin(names []string) (err error) {
 	return
 }
 
+func (p *PluginManager) writeFileLocally(statusCode int, body []byte, name string) (err error) {
+	if statusCode != 200 {
+		err = fmt.Errorf("unexpected status code: %d", statusCode)
+		if p.Debug {
+			ioutil.WriteFile(debugLogFile, body, 0664)
+		}
+		return err
+	}
+
+	err = ioutil.WriteFile(fmt.Sprintf("%s.hpi", name), body, 0644)
+	return err
+}
+
+// InstallPluginWithVersion install a plugin by name & version
+func (p *PluginManager) installPluginWithVersion(name string) (err error) {
+	pluginName := "%s.hpi"
+	pluginVersion := strings.Split(name, "@")
+	url := fmt.Sprintf("http://updates.jenkins-ci.org/download/plugins/%s/%s/%s.hpi",
```

Comment:
We cannot deal with the dependency of the plugin in this way. So, there're two suggestions below:

- resolve the dependency issues
- giving the warn tip about this

Reply:
There's a download method that exists. See also https://github.com/jenkins-zh/jenkins-cli/blob/master/client/pluginApi.go#L105

Reply:
@LinuxSuRen , i'm using the `download` api now.